src/index.ts

```diff
@@ -0,0 +1,275 @@
+	info,
+	setFailed,
+	endGroup,
+	startGroup,
+} from "@actions/core";
+import { execSync, exec } from "node:child_process";
+import { existsSync } from "node:fs";
+import * as path from "node:path";
+import * as util from "node:util";
+const execAsync = util.promisify(exec);
+
+const DEFAULT_WRANGLER_VERSION = "3.4.0";
+
+/**
+ * A configuration object that contains all the inputs & immutable state for the action.
+ */
+const config = {
+	WRANGLER_VERSION: getInput("wranglerVersion") || DEFAULT_WRANGLER_VERSION,
+	secrets: getMultilineInput("secrets"),
+	workingDirectory: checkWorkingDirectory(getInput("workingDirectory")),
+	CLOUDFLARE_API_TOKEN: getInput("apiToken"),
+	CLOUDFLARE_ACCOUNT_ID: getInput("accountId"),
+	ENVIRONMENT: getInput("environment"),
+	VARS: getMultilineInput("vars"),
+	COMMANDS: getMultilineInput("commands"),
+} as const;
+
+function getNpxCmd() {
+	return process.env.RUNNER_OS === "Windows" ? "npx.cmd" : "npx";
+}
+
+async function main() {
+	installWrangler();
+	authenticationSetup();
+	await execCommands(getMultilineInput("preCommands"), "Pre");
+	await uploadSecrets();
+	await wranglerCommands();
+	await execCommands(getMultilineInput("postCommands"), "Post");
+}
+
+function checkWorkingDirectory(workingDirectory = ".") {
+	try {
+		const normalizedPath = path.normalize(workingDirectory);
+		if (existsSync(normalizedPath)) {
+			return normalizedPath;
+		} else {
+			setFailed(`🚨 Directory ${workingDirectory} does not exist.`);
+		}
+	} catch (error) {
+		setFailed(
+			`🚨 While checking/creating directory ${workingDirectory} received ${error}`,
+		);
+	}
+}
+
+function installWrangler() {
+	if (config["WRANGLER_VERSION"].startsWith("1")) {
+		setFailed(
+			`🚨 Wrangler v1 is no longer supported by this action. Please use major versions 2 or 3`,
+		);
+	}
+	startGroup("📥 Installing Wrangler");
+	const command = `npm install wrangler@${config["WRANGLER_VERSION"]}`;
+	info(`Running Command: ${command}`);
+	execSync(command, { cwd: config["workingDirectory"], env: process.env });
+	info(`✅ Wrangler installed`);
+	endGroup();
+}
+
+function authenticationSetup() {
+	startGroup("🔐 Authenticating with Cloudflare");
+	try {
+		process.env.CLOUDFLARE_API_TOKEN = config["CLOUDFLARE_API_TOKEN"];
+		process.env.CLOUDFLARE_ACCOUNT_ID = config["CLOUDFLARE_ACCOUNT_ID"];
+		info(`Authentication process initiated with - API Token`);
+	} catch (error) {
+		setFailed(
+			`Authentication details were not found. Please input an 'apiToken' to the action.`,
+		);
+	}
+	endGroup();
+}
+
+async function execCommands(commands: string[], cmdType: string) {
+	if (!commands.length) {
+		return;
+	}
+
+	startGroup(`🚀 ${cmdType} Commands Group`);
+
+	const arrPromises = commands.map(async (command) => {
+		const npxCmd = command.startsWith("wrangler")
+			? `${getNpxCmd()} ${command}`
+			: command;
+
+		info(`🚀 Executing command: ${npxCmd}`);
+
+		await execAsync(npxCmd, {
+			cwd: config["workingDirectory"],
+			env: process.env,
+		});
```

Comment:
This is inside of a map that is creating an array of Promises that is passed into the Promise.all at line 107, so we definitely want it to be Async

Reply:
returning a promise is effectively the same as declaring the function `async` (except you'd get a `Promise<ChildProcess>` instead of a `Promise<void>` since you're not returning the `await execAsync()`). not a big deal either way.